# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/ffvpx/libavcodec/vp8data.h
# Commit: 186e2a334f2d
# Full Hash: 186e2a334f2deb37b29f44b28bc8c83babeeb9a4
# Author: Paul Adenot <paul@paul.cx>
# Date: 2024-02-09 09:37:17
# Regressor Bug: 1877221
# File Overlap Count: 1
# Description:
#   Bug 1877221 - Pull in new ffmpeg code, reapply patch. r=media-playback-reviewers,chunmin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D199909
# ==============================================================================

diff -r c5920de78d16 -r 186e2a334f2d media/ffvpx/libavcodec/vp8data.h
--- a/media/ffvpx/libavcodec/vp8data.h	Thu Feb 08 10:39:53 2024 +0000
+++ b/media/ffvpx/libavcodec/vp8data.h	Thu Feb 08 10:39:54 2024 +0000
@@ -513,176 +513,7 @@
     },
 };
 
-static const uint8_t vp8_token_update_probs[4][8][3][NUM_DCT_TOKENS - 1] = {
-    {
-        {
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-        },
-        {
-            { 176, 246, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 223, 241, 252, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 249, 253, 253, 255, 255, 255, 255, 255, 255, 255, 255 },
-        },
-        {
-            { 255, 244, 252, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 234, 254, 254, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 253, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-        },
-        {
-            { 255, 246, 254, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 239, 253, 254, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 254, 255, 254, 255, 255, 255, 255, 255, 255, 255, 255 },
-        },
-        {
-            { 255, 248, 254, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 251, 255, 254, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-        },
-        {
-            { 255, 253, 254, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 251, 254, 254, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 254, 255, 254, 255, 255, 255, 255, 255, 255, 255, 255 },
-        },
-        {
-            { 255, 254, 253, 255, 254, 255, 255, 255, 255, 255, 255 },
-            { 250, 255, 254, 255, 254, 255, 255, 255, 255, 255, 255 },
-            { 254, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-        },
-        {
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-        },
-    },
-    {
-        {
-            { 217, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 225, 252, 241, 253, 255, 255, 254, 255, 255, 255, 255 },
-            { 234, 250, 241, 250, 253, 255, 253, 254, 255, 255, 255 },
-        },
-        {
-            { 255, 254, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 223, 254, 254, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 238, 253, 254, 254, 255, 255, 255, 255, 255, 255, 255 },
-        },
-        {
-            { 255, 248, 254, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 249, 254, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-        },
-        {
-            { 255, 253, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 247, 254, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-        },
-        {
-            { 255, 253, 254, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 252, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-        },
-        {
-            { 255, 254, 254, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 253, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-        },
-        {
-            { 255, 254, 253, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 250, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 254, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-        },
-        {
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-        },
-    },
-    {
-        {
-            { 186, 251, 250, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 234, 251, 244, 254, 255, 255, 255, 255, 255, 255, 255 },
-            { 251, 251, 243, 253, 254, 255, 254, 255, 255, 255, 255 },
-        },
-        {
-            { 255, 253, 254, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 236, 253, 254, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 251, 253, 253, 254, 254, 255, 255, 255, 255, 255, 255 },
-        },
-        {
-            { 255, 254, 254, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 254, 254, 254, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-        },
-        {
-            { 255, 254, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 254, 254, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 254, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-        },
-        {
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 254, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-        },
-        {
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-        },
-        {
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-        },
-        {
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-        },
-    },
-    {
-        {
-            { 248, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 250, 254, 252, 254, 255, 255, 255, 255, 255, 255, 255 },
-            { 248, 254, 249, 253, 255, 255, 255, 255, 255, 255, 255 },
-        },
-        {
-            { 255, 253, 253, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 246, 253, 253, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 252, 254, 251, 254, 254, 255, 255, 255, 255, 255, 255 },
-        },
-        {
-            { 255, 254, 252, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 248, 254, 253, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 253, 255, 254, 254, 255, 255, 255, 255, 255, 255, 255 },
-        },
-        {
-            { 255, 251, 254, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 245, 251, 254, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 253, 253, 254, 255, 255, 255, 255, 255, 255, 255, 255 },
-        },
-        {
-            { 255, 251, 253, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 252, 253, 254, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 255, 254, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-        },
-        {
-            { 255, 252, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 249, 255, 254, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 255, 255, 254, 255, 255, 255, 255, 255, 255, 255, 255 },
-        },
-        {
-            { 255, 255, 253, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 250, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-        },
-        {
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 254, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-            { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 },
-        },
-    },
-};
+extern const uint8_t ff_vp8_token_update_probs[4][8][3][11];
 
 static const uint8_t vp8_dc_qlookup[VP8_MAX_QUANT + 1] = {
       4,   5,   6,   7,   8,   9,  10,  10,  11,  12,  13,  14,  15,  16,  17,  17,